pkg/container-collection/container-collection.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package containercollection provides the ContainerCollection struct to keep
+// track of the set of running containers and primitives to query that set with
+// various criteria.
+//
+// It is used by the Gadget Tracer Manager to keep track of containers part of
+// Kubernetes pods and by Local Gadget Manager to keep track of containers on a
+// Linux system.
+package containercollection
+
+import (
+	"sync"
+
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/pubsub"
+)
+
+// ContainerCollection holds a set of containers. It can be embedded as an
+// anonymous struct to help other structs implement the ContainerResolver
+// interface. For this reason, some methods are prefixed with 'Container' to
+// make clear that
+type ContainerCollection struct {
+	// Keys:   Container ID
+	// Values: pb.ContainerDefinition
```

Comment:
I'm a bit confused about saving the structure or saving a pointer to it. Wouldn't it better to save the pointer and pass pointers everywhere? 

also I remember I discussed with @blanquicet about a warning that was generated because we're copying pb.ContainerDefinition and it contains a lock. Do you remember what our conclusion was?

Reply:
We concluded that saving pointers would be the best short-term solution. However, we all agreed that the final solution should be a refactoring that removes the use of an internal data structure that would remove the `pb.*` dependency.

Reply:
As opposed to standard Go maps, `sync.Map` uses `interface{}` so it uses pointers implicitly.

Reply:
Mmmm if it is a pointer why we don't do `value.(*pb.ContainerDefinition)` and same for `Store`? Sorry but I get confused when golang hides the pointer stuff for me 😅 .